pkg/services/guardian/accesscontrol_guardian.go

```diff
@@ -107,50 +95,72 @@ func NewAccessControlDashboardGuardianByUID(
 			user:             user,
 			ac:               ac,
 			dashboardService: dashboardService,
+			folderService:    folderService,
 		},
 		dashboard: dashboard,
 	}, nil
 }
 
-// NewAccessControlDashboardGuardianByDashboard creates a dashboard guardian by the provided dashboard.
-// This constructor should be preferred over the other two if the dashboard in available
-// since it avoids querying the database for fetching the dashboard.
-func NewAccessControlDashboardGuardianByDashboard(
-	ctx context.Context, cfg *setting.Cfg, dashboard *dashboards.Dashboard, user identity.Requester,
-	ac accesscontrol.AccessControl, dashboardService dashboards.DashboardService,
+// NewAccessControlDashboardGuardianByFolderUID creates a folder guardian by the provided folderUID.
+func NewAccessControlDashboardGuardianByFolderUID(
```

Comment:
Naming is weird for this one. Would `NewAccessControlFolderGuardianByUID` make more sense?

Reply:
Definitely - good call. Updated!